### DISCUSSION TITLE
Compiler Plugin: How to make the task fail when issues are found

### DISCUSSION THREAD

**Discussion by JanMalch** [detekt/detekt#6388] [Q&A]
I would like to use detekt with the compiler plugin in the git pre-push hook. Currently, when issues are found the build still succeeds. What's the best way to make the build fail in that case?
My current solution is to `grep` the output with `"Analysis failed with "`. I also thought about using a txt report and checking if that file contains any lines. But of course all that feels rather hacky.

**Comment by cortinico** [comment]
> Currently, when issues are found the build still succeeds. What's the best way to make the build fail in that case?

Have you tried using the `allWarningsAsErrors` from the Kotlin Gradle Plugin?
See https://kotlinlang.org/docs/gradle-compiler-options.html#attributes-common-to-jvm-js-and-js-dce
